tests/bluetooth/bsim_bt/bsim_test_audio/src/broadcast_source_test.c

```diff
@@ -96,13 +96,57 @@ static struct bt_audio_stream_ops stream_ops = {
 	.sent = sent_cb
 };
 
-static int setup_extended_adv(const struct bt_audio_broadcast_source *source,
+static int setup_broadcast_source(struct bt_audio_broadcast_source **source)
+{
+	struct bt_codec_data bis_codec_data = BT_CODEC_DATA(BT_CODEC_CONFIG_LC3_FREQ,
+							    BT_CODEC_CONFIG_LC3_FREQ_16KHZ);
+	struct bt_audio_broadcast_source_stream_param
+		stream_params[ARRAY_SIZE(broadcast_source_streams)];
+	struct bt_audio_broadcast_source_subgroup_param
+		subgroup_params[CONFIG_BT_AUDIO_BROADCAST_SRC_SUBGROUP_COUNT];
+	struct bt_audio_broadcast_source_create_param create_param;
+	int err;
+
+
+	(void)memset(broadcast_source_streams, 0,
+		     sizeof(broadcast_source_streams));
+
+	for (size_t i = 0; i < ARRAY_SIZE(stream_params); i++) {
+		stream_params[i].stream = &broadcast_source_streams[i];
+		bt_audio_stream_cb_register(stream_params[i].stream,
+					    &stream_ops);
+		stream_params[i].data_count = 1U;
+		stream_params[i].data = &bis_codec_data;
+	}
+
+	for (size_t i = 0U; i < ARRAY_SIZE(subgroup_params); i++) {
+		subgroup_params[i].stream_count = 1U;
+		subgroup_params[i].stream_params = &stream_params[i];
+		subgroup_params[i].codec = &preset_16_2_1.codec;
+	}
+
+	create_param.subgroup_count = ARRAY_SIZE(subgroup_params);
+	create_param.subgroup_params = subgroup_params;
+	create_param.qos = &preset_16_2_2.qos;
+
+	printk("Creating broadcast source with %zu subgroups and %zu streams\n",
+	       ARRAY_SIZE(subgroup_params), ARRAY_SIZE(stream_params));
+	err = bt_audio_broadcast_source_create(&create_param, source);
+	if (err != 0) {
+		printk("Unable to create broadcast source: %d\n", err);
+		return err;
+	}
+
+	return 0;
+}
+
+static int setup_extended_adv(struct bt_audio_broadcast_source *source,
 			      struct bt_le_ext_adv **adv)
 {
 	/* Broadcast Audio Streaming Endpoint advertising data */
 	NET_BUF_SIMPLE_DEFINE(ad_buf,
 			      BT_UUID_SIZE_16 + BT_AUDIO_BROADCAST_ID_SIZE);
-	NET_BUF_SIMPLE_DEFINE(base_buf, 64);
+	NET_BUF_SIMPLE_DEFINE(base_buf, 128);
```

Comment:
magic number

Reply:
It's just a buffer large enough to hold some data. 

Would you like me to modify the value to the specific exact value needed (which would then need an update every time we modify the number of streams, the codec configuration or the metadata)?

Reply:
You're right, it does not make sense